libsignal-protocol/tests/libsignal-protocol-c-tests.rs

```diff
@@ -400,3 +411,124 @@ fn test_optional_one_time_pre_key() {
     let has_pre_key_id = incoming_message.has_pre_key_id();
     assert!(!has_pre_key_id);
 }
+
+#[test]
+fn test_decrypt() {
+    let bob_address = Address::new("+14152222222", 1);
+    let alice_address = Address::new("+14157777777", 1);
+    let ctx = mock_ctx();
+
+    // Create Alice's data store and session builder
+    let alice_identity = sig::generate_identity_key_pair(&ctx).unwrap();
+    let alice_store = sig::store_context(
+        &ctx,
+        InMemoryPreKeyStore::default(),
+        InMemorySignedPreKeyStore::default(),
+        InMemorySessionStore::default(),
+        InMemoryIdentityKeyStore::new(
+            sig::generate_registration_id(&ctx, 0).unwrap(),
+            &alice_identity,
+        ),
+    )
+    .unwrap();
+    let alice_session_builder =
+        sig::session_builder(&ctx, &alice_store, &bob_address);
+
+    // Create Bob's data store and pre key bundle
+    let registration_id = sig::generate_registration_id(&ctx, 0).unwrap();
+    let bob_identity_key_pair = sig::generate_identity_key_pair(&ctx).unwrap();
+    let bob_store = sig::store_context(
+        &ctx,
+        InMemoryPreKeyStore::default(),
+        InMemorySignedPreKeyStore::default(),
+        InMemorySessionStore::default(),
+        InMemoryIdentityKeyStore::new(registration_id, &bob_identity_key_pair),
+    )
+    .unwrap();
+
+    let bob_local_registration_id = bob_store.registration_id().unwrap();
```

Comment:
What's the difference between `bob_local_registration_id` and `registration_id`? Are they actually the same ID?

Reply:
They are. Fixed in 34149a116.